### ISSUE TITLE
Extension blocks: extension method called from constructor is marked as unused

### ISSUE THREAD

**Issue by jhinder** [dotnet/roslyn#80645]
**Version Used**: VS 18.0.0 Insiders [11109.219]

**Steps to Reproduce**:
Observe the method declaration for `M`.
```csharp
static class D
{
    class C
    {
        public C()
        {
            this.M();
        }
    }

    extension (C c)
    {
        private void M()
        {
        }
    }
}
```

**Expected Behavior**:
Since the method is used, it is not marked as unused.

**Actual Behavior**:
The method M is greyed out and IDE0051 is reported. If you select the code fix "Remove unused member", the code breaks.

Note: if you declare the extension method in the "old way" (i.e. `private static void M(this C c)`), this will not occur.

In the following screenshot, you can even see that the syntax highlighter shows that the method is actually used.
<img width="756" height="530" alt="Image" src="https://github.com/user-attachments/assets/df22efd5-e1c4-4eef-996f-df6c89344a67" />

**Comment by jhinder** [comment]
OK, so it's apparently not specific to constructors, here's another example I found while playing around, where both `M` methods are shown as "unused" despite being used:

```csharp
internal static class Program
{
    private static void Main(string[] args)
    {
        _ = from a in args.M()
            let v = a.M()
            select v;
    }

    extension<T> (T s)
    {
        private T M() => s;
    }

    extension<T> (T[] v)
    {
        private T[] M() => v;
    }
}
```
